### PR TITLE
DROID-2511 Protocol | MW 0.34.0-rc2

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt [hunk 1]
```diff
@@ -1193,11 +1193,11 @@ class Middleware @Inject constructor(
 
     @Throws(Exception::class)
     fun objectRemoveCover(ctx: String): Payload {
-        val coverIdDetail = Rpc.Object.SetDetails.Detail(
+        val coverIdDetail = anytype.model.Detail(
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt:1196]: consider using typealias here

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt [hunk 2]
```diff
@@ -1193,11 +1193,11 @@ class Middleware @Inject constructor(
...
             key = coverIdKey,
             value_ = null
         )
-        val coverTypeDetail = Rpc.Object.SetDetails.Detail(
+        val coverTypeDetail = anytype.model.Detail(
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt:1200]: consider using typealias here

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt [hunk 3]
```diff
@@ -1213,11 +1213,11 @@ class Middleware @Inject constructor(
 
     @Throws(Exception::class)
     fun objectRemoveIcon(ctx: String): Payload {
-        val imageDetail = Rpc.Object.SetDetails.Detail(
+        val imageDetail = anytype.model.Detail(
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt:1216]: consider using typealias here

**Review by uburoiubu — APPROVED** [review]

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt (hunk 2 above)
Comment (konstantiniiv) [reply middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt]: done

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt (hunk 3 above)
Comment (konstantiniiv) [reply middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt]: done

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt (hunk 1 above)
Comment (konstantiniiv) [reply middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt]: done

**Pull request merged**